src/bcf/report/static_reader.rs

```diff
@@ -0,0 +1,170 @@
+    flags_map.insert(0x1, FLAG_1);
+    flags_map.insert(0x2, FLAG_2);
+    flags_map.insert(0x4, FLAG_3);
+    flags_map.insert(0x8, FLAG_4);
+    flags_map.insert(0x10, FLAG_5);
+    flags_map.insert(0x20, FLAG_6);
+    flags_map.insert(0x40, FLAG_7);
+    flags_map.insert(0x80, FLAG_8);
+    flags_map.insert(0x100, FLAG_9);
+    flags_map.insert(0x200, FLAG_10);
+    flags_map.insert(0x400, FLAG_11);
+    flags_map.insert(0x800, FLAG_12);
+
+    for (flag, text) in flags_map {
+        for f in flag_vec.clone() {
+            if (flag & f) == flag {
+                string_map.insert(flag, text);
+            }
+        }
+    }
+
+    string_map
+}
+
+fn calc_rows(
+    reads: Vec<AlignmentNucleobase>,
+    matches: Vec<AlignmentMatch>,
+) -> (Vec<StaticAlignmentNucleobase>, Vec<StaticAlignmentMatch>) {
+    let mut row_ends = vec![0; 30];
+
+    let mut read_names: BTreeMap<String, u8> = BTreeMap::new();
+
+    let mut reads_wr: Vec<StaticAlignmentNucleobase> = Vec::new();
+    let mut matches_wr: Vec<StaticAlignmentMatch> = Vec::new();
+
+    for r in matches {
+        let mut row: u8 = 0;
+
+        if read_names.contains_key(&r.name) {
+            row = *read_names.get(&r.name).unwrap();
+        } else {
+            for i in 1..30 {
+                if r.read_start > row_ends[i] {
+                    row = i as u8;
+                    row_ends[i] = r.read_end;
+                    read_names.insert(r.name.clone(), i as u8);
+                    break;
+                }
+            }
+        }
+
+        let f = decode_static_flags(r.flags);
+
+        let base = StaticAlignmentMatch {
+            marker_type: r.marker_type,
+            start_position: r.start_position,
+            end_position: r.end_position,
+            flags: f,
+            name: r.name,
+            row: row,
+        };
+
+        matches_wr.push(base);
+    }
+
+    for r in reads {
+        let mut row: u8 = 0;
+
+        if read_names.contains_key(&r.name) {
+            row = *read_names.get(&r.name).unwrap();
+        } else {
+            for i in 1..30 {
+                if r.read_start > row_ends[i] {
+                    row = i as u8;
+                    row_ends[i] = r.read_end;
+                    read_names.insert(r.name.clone(), i as u8);
+                    break;
+                }
+            }
+        }
+
+        let f = decode_static_flags(r.flags);
+
+        let base = StaticAlignmentNucleobase {
+            marker_type: r.marker_type,
+            bases: r.bases,
+            start_position: r.start_position,
+            end_position: r.end_position,
+            flags: f,
+            name: r.name,
+            row: row,
+        };
+
+        reads_wr.push(base);
+    }
+
+    (reads_wr, matches_wr)
+}
+
+pub fn get_static_reads(
+    path: &Path,
+    fasta_path: &Path,
+    chrom: String,
+    from: u64,
+    to: u64,
+) -> (Vec<StaticAlignmentNucleobase>, Vec<StaticAlignmentMatch>) {
```

Comment:
What is the difference between an AlignmentNucleobase and an AlignmentMatch?

Reply:
The `AlignmentNucleobase` struct represents a single base that is not matching the reference and is basically a mismatch. The `AlignmentMatch` struct is the opposite and represents an interval of bases that match the reference.